acados/ocp_nlp/ocp_nlp_constraints_bghp.c

```diff
@@ -1073,12 +1088,43 @@ void ocp_nlp_constraints_bghp_update_qp_matrices(void *config_, void *dims_, voi
         model->p->evaluate(model->p, ext_fun_type_in, ext_fun_in, ext_fun_type_out, ext_fun_out);
 
         // SCQP Hessian
-        double lam = blasfeo_dvecex1(memory->lam, 2 * (nb + ng) + nh) -
-                     blasfeo_dvecex1(memory->lam, nb + ng);
-
-        blasfeo_dsyrk_ln(nx + nu, np, 2 * lam, &work->jacobian_quadratic, 0, 0,
-                         &work->jacobian_quadratic, 0, 0, 1.0, memory->RSQrq, 0, 0, memory->RSQrq,
-                         0, 0);
+        
+        for (int i = 0; i < nh; i++) { 
+            // TODO(andrea): @giaf: if the lower bound is active we shouldn't consider its contribution, right?
+            // Removing this.
+            // double lam_i = blasfeo_dvecex1(memory->lam, 2 * (nb + ng) + nh) -
+            //              blasfeo_dvecex1(memory->lam, nb + ng);
+
+            double lam_i = blasfeo_dvecex1(memory->lam, 2 * (nb + ng) + nh + i);
+
+            // printf("lam_i = %f", lam_i);
+
+            // andrea: this did not take into account the Hessian of the convex outer part
+            // blasfeo_dsyrk_ln(nx + nu, np, 2 * lam, &work->jacobian_quadratic, 0, 0,
+            //                  &work->jacobian_quadratic, 0, 0, 1.0, memory->RSQrq, 0, 0, memory->RSQrq,
+            //                  0, 0);
+            
+            // printf("memory->RSQrq:\n");
+            // blasfeo_dgese(0.0, nv, nv, memory->RSQrq, 0, 0);
+            // blasfeo_print_dmat(nv, nv, memory->RSQrq, 0, 0);
+
+            // printf("tmp_np_nh_np:\n");
+            // blasfeo_print_dmat(np * nh, np, &work->tmp_np_nh_np, 0, 0);
+
+            blasfeo_dgemm_nt(nv, np, np, lam_i, &work->jacobian_quadratic, 
```

Comment:
using `blasfeo_dpotrf_l` + `blasfeo_dtrmm_rlnn` is cheaper if roughly `nr<=nx+nu`

Reply:
the same algorithm is used in the precompute of the linear least squares, for reference